system/handlers/admin/DataManager.cfc

```diff
@@ -489,6 +489,40 @@ component extends="preside.system.base.AdminHandler" {
 		}
 	}
 
+	public void function flagRecordAction( event, rc, prc, args={} ) {
+		var objectName = rc.object ?: "";
+		var recordId   = rc.id     ?: "";
+		var resultView = rc.result ?: "";
+		var successUrl = event.buildAdminLink( objectName=objectName );
+		var updated    = false;
+
```

Comment:
There should be a permissions check here. Can also add a `isFlaggingEnabled()` check at the same point.

Reply:
added edit permission check and object flagging enabled check